velox/connectors/hive/storage_adapters/gcs/GCSFileSystem.cpp

```diff
@@ -72,7 +72,12 @@ class GCSReadFile final : public ReadFile {
 
   // Gets the length of the file.
   // Checks if there are any issues reading the file.
-  void initialize() {
+  void initialize(const filesystems::FileOptions& options) {
+    if (options.values.count("fileSize") > 0) {
+      length_ = !options.values.at("fileSize").empty()
```

Comment:
Let's add ` static constexpr folly::StringPiece kFileSize{"file-size"};` to the `FileSize` class.
We should throw an error if `options.values.at("fileSize").empty()` is true as it is a user error.

Reply:
I see what you mean, updated. Thanks!

Reply:
Yes, `FileOptions` struct. You got this right!